File: app/src/main/java/com/practice/nycschools/model/DataViewModel.java
package com.practice.nycschools.model;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

public class DataViewModel extends ViewModel {

    // Create a LiveData with a String
    private MutableLiveData<List<NYCListClass>> currentData;

    public MutableLiveData<List<NYCListClass>> getCurrentData() {
        if (currentData == null) {
            currentData = new MutableLiveData<>();
        }
        return currentData;
    }

    private MutableLiveData<List<SchoolClass>> currentSchool;

    public MutableLiveData<List<SchoolClass>> getCurrentSchool() {
        if (currentSchool == null) {
            currentSchool = new MutableLiveData<>();
        }
        return currentSchool;
    }
}

File: app/src/main/java/com/practice/nycschools/model/DataViewModel.kt
package com.practice.nycschools.model

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class DataViewModel : ViewModel() {

    var currentData: MutableLiveData<List<NYCListClass>> = MutableLiveData()
        private set

    var currentSchool: MutableLiveData<List<SchoolClass>> = MutableLiveData()
        private set
}